Feature/cax 1 prs@189 Document how to debug connectors deployed in Kubernetes and how to clear the db. (#99)
catenax/tractusx: coreservices/partsrelationshipservice/connectors_scripts/README.md

First, run:

```bash
az aks get-credentials --resource-group <resource-group> --name <aks-cluster-where-connectors-are-deployed>
kubectl scale deployment consumer-dataspace-connector --replicas=0
kubectl scale deployment producer-dataspace-connector --replicas=0
kubectl scale sts consumer-postgresql --replicas=0
kubectl scale sts producer-postgresql --replicas=0
```

Run `kubectl get pods` and make sure that no pods are running to continue.

```bash
kubectl delete pvc data-consumer-postgresql-0
kubectl delete pvc data-producer-postgresql-0
kubectl scale sts consumer-postgresql --replicas=1
kubectl scale sts consumer-postgresql --replicas=1
kubectl scale deployment consumer-dataspace-connector --replicas=1
kubectl scale deployment producer-dataspace-connector --replicas=1
```

## Debug the connectors deployed in kubernetes

It is possible to debug connectors that running in Kubernetes.
For that, you need to redeploy the connectors with remote debug enabled.

The connector helm chart is located in the [/connector/helm-chart](/connector/helm-chart) folder.

> Configurations to deploy helm chart connectors in DEV are located in /connector/landscape/dev006/.

Reviewer: no link here?

Author: Yes because these paths were not in the repo yet :).
But I see Ivan merged his PRs will fix it.

Configurations to deploy helm chart connectors in INT environment are located in /connector/landscape/int/.
The configurations for the PRS consumer are in the prs-query folder and configurations for producer are in the prs-upload folder.

To deploy a connector with remoteDebugEnabled, set remoteDebugEnabled to true in your values.yml.
Then redeploy the connector with helm upgrade. Instructions to deploy the helm chart are in [/connector/landscape/README.md](/connector/landscape/README.md)

Run the following command:

```bash

> kubectl port-forward $(kubectl get pod -l app.kubernetes.io/instance=<release-name> -o jsonpath="{.items[0].metadata.name}") 5005

Reviewer: I think `kubectl port-forward deployments/prs 5005` should work too and it is a bit simpler

```

> Then open the [Dataspace connector project](https://github.com/International-Data-Spaces-Association/DataspaceConnector) in intellij.

Reviewer: ```suggestion
Then open the [Dataspace connector project](https://github.com/International-Data-Spaces-Association/DataspaceConnector) in your IDE.
```

> Go to edit configuration > add configuration > Remote JVM debug.

Reviewer: I would make this line a bit more generic that could work for any IDE like `Add a Remote JVM debugging configuration`

[resourceapi.py](https://github.com/International-Data-Spaces-Association/DataspaceConnector/blob/main/scripts/tests/resourceapi.py) and [idsapi.py](https://github.com/International-Data-Spaces-Association/DataspaceConnector/blob/main/scripts/tests/idsapi.py) have been taken in the [Dataspace connector repository](https://github.com/International-Data-Spaces-Association/DataspaceConnector).
[create_catalog_and_artifact.py](./create_catalof_and_artifact.py) and [consume_artifact.py](./negotiate_contract_and_consume_artifact.py) are based on [a script from the DataspaceConnector repository](https://github.com/International-Data-Spaces-Association/DataspaceConnector/blob/main/scripts/tests/contract_negotation_allow_access.py).
[create_catalog_and_artifact.py](./create_catalof_and_artifact.py) creates a catalog and an artifact accessible via an access url (our PRS api in our case).

> [consume_artifact.py](./negotiate_contract_and_consume_artifact.py) Finds the first artifact of the first catalog accessible and tries to access the artifact data by calling the access_url of the artifact. It takes the first artifact for simplicity as we are supposed to register only one artifact. You can specify the pathparams and query params that needs to be appended to the access url to access a resource.

Reviewer 2: nit: 
```suggestion
[consume_artifact.py](./negotiate_contract_and_consume_artifact.py) Finds the first artifact of the first catalog accessible and tries to access the artifact data by calling the access_url of the artifact. It takes the first artifact for simplicity as we are supposed to register only one artifact. You can specify the pathparams and query params that need to be appended to the access url to access a resource.
```

## Cleanup the data of the connectors

If you want to start from a fresh state and remove catalogs, resources, everything from postgres you can run the following commands to clean the postgres database.

First, run:

```bash
az aks get-credentials --resource-group <resource-group> --name <aks-cluster-where-connectors-are-deployed>
kubectl scale deployment consumer-dataspace-connector --replicas=0
kubectl scale deployment producer-dataspace-connector --replicas=0
kubectl scale sts consumer-postgresql --replicas=0
kubectl scale sts producer-postgresql --replicas=0
```

Run `kubectl get pods` and make sure that no pods are running to continue.

```bash
kubectl delete pvc data-consumer-postgresql-0
kubectl delete pvc data-producer-postgresql-0
kubectl scale sts consumer-postgresql --replicas=1

> kubectl scale sts consumer-postgresql --replicas=1

Reviewer 2: ```suggestion
kubectl scale sts producer-postgresql --replicas=1
```

Author: Good catch
